server/server.go

```diff
@@ -72,11 +72,12 @@ func NewProtocolServer(certFile, keyFile string) (*Server, error) {
 }
 
 // Start configures and starts the protocol server at the requested port.
-func (s *Server) Start(port int) {
+func (s *Server) Start(port int) error {
 	connStr := fmt.Sprintf(":%d", port)
 	listener, err := net.Listen("tcp", connStr)
 	if err != nil {
 		logger.Criticalf("Could not connect: %v", err)
+		return err
```

Comment:
I would perfer if you just return the error and leave it to the caller to print it:

```go
if err != nil {
  return fmt.Errorf("Could not connect: %v", err)
}
```

Currently the error gets printed twice:

```
2017/09/13 08:03:09 [server] 08:03:09.498 Start ▶ CRIT 006 Could not connect: listen tcp :7007: bind: address already in use
listen tcp :7007: bind: address already in use
```

I remember us having discussion about this in past already, but I can't recall what the final decision was. If we agreed to both log and return error, then please just ignore this comment.

Reply:
We did discuss this previously and decided in favor of what you propose.
Done.